File: app/src/main/java/com/riders/thelab/ui/weather/WeatherForecastAdapter.java
package com.riders.thelab.ui.weather;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.riders.thelab.R;
import com.riders.thelab.data.remote.dto.weather.DailyWeather;

import org.jetbrains.annotations.NotNull;

import java.util.List;

public class WeatherForecastAdapter extends RecyclerView.Adapter<WeatherForecastViewHolder> {

    private final Context context;
    private final List<DailyWeather> dailyWeatherList;

    public WeatherForecastAdapter(Context context, List<DailyWeather> dailyWeatherList) {
        this.context = context;
        this.dailyWeatherList = dailyWeatherList;
    }

    @NonNull
    @NotNull
    @Override
    public WeatherForecastViewHolder onCreateViewHolder(@NonNull @NotNull ViewGroup parent, int viewType) {
        return new WeatherForecastViewHolder(
                context,
                LayoutInflater
                        .from(context)
                        .inflate(R.layout.row_weather_forecast, parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull @NotNull WeatherForecastViewHolder holder, int position) {
        DailyWeather dailyWeather = dailyWeatherList.get(position);

        holder.bindData(dailyWeather);
    }

    @Override
    public int getItemCount() {
        if (!dailyWeatherList.isEmpty())
            return dailyWeatherList.size();

        return 0;
    }
}

File: app/src/main/java/com/riders/thelab/ui/weather/WeatherForecastAdapter.kt
package com.riders.thelab.ui.weather

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.riders.thelab.R
import com.riders.thelab.data.remote.dto.weather.DailyWeather

class WeatherForecastAdapter constructor(
    val context: Context,
    val dailyWeatherList: List<DailyWeather>
) :
    RecyclerView.Adapter<WeatherForecastViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): WeatherForecastViewHolder {
        return WeatherForecastViewHolder(
            context,
            LayoutInflater
                .from(context)
                .inflate(R.layout.row_weather_forecast, parent, false)
        )
    }

    override fun onBindViewHolder(holder: WeatherForecastViewHolder, position: Int) {
        val dailyWeather: DailyWeather = dailyWeatherList[position]
        holder.bindData(dailyWeather)
    }

    override fun getItemCount(): Int {
        return if (dailyWeatherList.isNotEmpty()) dailyWeatherList.size else 0
    }
}